homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,185 @@
+
+from aiopylgtv import PyLGTVPairException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_CLIENT_SECRET, CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from . import CannotConnect, async_control_connect
+from .const import CONF_ON_ACTION, CONF_SOURCES, DEFAULT_NAME, DOMAIN, ON_ACTION_DOCS
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """WebosTV configuration flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize workflow."""
+        self._user_input = {}
+        self._force_pairing = False
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        self._force_pairing = True
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            self._user_input = user_input
+            self.context["title_placeholders"] = {
+                "name": user_input.get(CONF_NAME, CONF_HOST)
+            }
+            if self._force_pairing:
+                return await self.async_step_pairing({})
+
+            return await self.async_step_pairing()
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_pairing(self, user_input=None):
+        """Display pairing form."""
+        errors = {}
+        if user_input is not None:
+            try:
+                client = await async_control_connect(self._user_input[CONF_HOST], None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            else:
+                return await self.async_step_register(self._user_input, client)
+
+        return self.async_show_form(
+            step_id="pairing",
+            errors=errors,
+        )
+
+    async def async_step_register(self, user_input, client=None):
+        """Register entity."""
+        if not client.is_registered():
+            return await self.async_step_user()
+
+        if client.client_key is None:
+            return self.async_abort(reason="client_key_not_found")
+
+        await self.async_set_unique_id(client.software_info["device_id"])
+        self._abort_if_unique_id_configured()
```

Comment:
We should update the ip address if its different since we just validated the device id

```suggestion
        self._abort_if_unique_id_configured(updates={CONF_HOST: self._user_input[CONF_HOST]})
```